### DISCUSSION TITLE
Python mtest module not working as expected?

### DISCUSSION THREAD

**Discussion by mhpierre** [thelfer/tfel#563] [Q&A] [answered]
Hi,
I'm trying to implement more advanced things in `mtest`, namely finding the right increment of driving variables to ensure some condition on some internal state variables. This would require accessing the internal state variable and using something like a Newton algorithm to find the right increments.
The `python` module works fine for me in a simple case (using `MTest.execute()`), however when I try to get under the hood I'm faced with some problems.
Firstly, the `printOutput` function seems to yield no output. The file is empty after calling it.
Secondly, I can't seem to access the underlying elements of the `MTestCurrentState` class.  Trying to access `s1` for instance , I get this error:
```
TypeError: No to_python (by-value) converter found for C++ type: tfel::math::vector<double>
```
I'm currently working with `tfel 4.2.0` against `python 3.11.6`.

Thanks in advance,
Maxime

**Comment by thelfer** [comment]
> irstly, the printOutput function seems to yield no output. The file is empty after calling it.

The attached example [test.zip](https://github.com/thelfer/tfel/files/15354527/test.zip) works fine. Would you test it ? 

```bash
$ mfront --obuild --interface=castem Tvergaard.mfront
$ python test.py
```

> Secondly, I can't seem to access the underlying elements of the MTestCurrentState class. Trying to access s1 for instance , I get this error

You need to import the `std` module firts

Best

**Reply by mhpierre** [reply]
Your example works fine indeed, there seems to be a weird interaction with jupyterlab where it doesn't print to the file but as a python script it works, hence my confusion.

> You need to import the std module firts

I had the `std` module imported already, I still have this error ...

**Reply by mhpierre** [reply]
For some reason, adding `import tfel.math` seems to do the trick.

**Reply by thelfer** [reply, marked as answer]
My bad, I misread the error message. I though we were taking `std::vector`. For `tfel::math::vector`, you indeed to need to import the `tfel.math` model
